src/library_syscall.js

```diff
@@ -1444,7 +1444,16 @@ function wrapSyscallFunction(x, library, isWasi) {
   var canThrow = library[x + '__nothrow'] !== true;
 #endif
 
-  var pre = '', post = '';
+  if (!library[x + '__deps']) library[x + '__deps'] = [];
+  library[x + '__deps'].push('$SYSCALLS');
+
+#if PURE_WASI
+  var pre = '\nif (!HEAPU8.byteLength) _emscripten_notify_memory_growth(0);\n'
```

Comment:
this deserves a comment. I'm guessing this is because we need to check for growth on each call, without a proper notification hook?